### PR TITLE
add simple stats method

### PR DESCRIPTION
**Pull request by walldiss**
Add simple method to expose basic metrics

### CODE REVIEW

**Review by distractedm1nd — APPROVED** [review]

File: dagstore.go [hunk 1]
```diff
@@ -519,6 +519,17 @@ func (d *DAGStore) GetShardInfo(k shard.Key) (ShardInfo, error) {
 	return info, nil
 }
 
+func (d *DAGStore) Stats() map[ShardState]int {
+	d.lk.RLock()
+	defer d.lk.RUnlock()
+
+	stats := make(map[ShardState]int)
```
Comment (Wondertan) [inline dagstore.go:526]: Prealloc

Comment (walldiss) [reply dagstore.go:526]: What do you think should be the capacity?

Comment (Wondertan) [reply dagstore.go:526]: Len(shards)

Comment (walldiss) [reply dagstore.go:526]: It is grouping values by ShardState, there will be very few elements inside this map

**Review by renaynay — APPROVED** [review]
Approved - re @Wondertan's comment, it doesn't make sense to prealloc the map is we do not know how many ShardStates will actually be present

**Pull request merged**